lib/cluster.js

```diff
@@ -110,12 +110,11 @@ function RoundRobinHandle(key, address, port, addressType, backlog, fd) {
   else
     this.server.listen(address);  // UNIX socket path.
 
-  var self = this;
-  this.server.once('listening', function() {
-    self.handle = self.server._handle;
-    self.handle.onconnection = self.distribute.bind(self);
-    self.server._handle = null;
-    self.server = null;
+  this.server.once('listening', () => {
+    this.handle = this.server._handle;
+    this.handle.onconnection = this.distribute.call(this);
```

Comment:
Yes, this should not be a `.call()`, otherwise the function will be immediately invoked.